# Grand Rapids Traffic Accident Project
## Part 1: Data Cleaning and Exploration

Created by: Kate Meredith
Date: 11.28.22

## Table of Contents

* [1. Background](#header1)
    * [1.1 Data Dictionary](#subheader11)
* [2. Importing Libraries](#header2)
* [3. Importing the Data](#header3)
* [4. Exploring the Data Composition](#header4)

## 1. Background <a class="anchor" id="header1"></a>

This data represents traffic accidents occuring in Grand Rapids, Michigan, USA from 2007 to 2017. The
data comes from [Grand Rapids Open Data](https://grdata-grandrapids.opendata.arcgis.com/datasets/grandrapids::cgr-crash-data/explore?location=0.000000%2C0.000000%2C2.62).

This [web page](https://services2.arcgis.com/L81TiOwAPO1ZvU9b/arcgis/rest/services/CGRCrashData/FeatureServer/0) provides an overview of the variables and their datatypes.

### 1.1 Data Dictionary <a class="anchor" id="subheader11"></a>

The dataset contains a tremendous amount of detail around the circumstances of the crash (142 variables). The data did not include a data dictionary, so the following provides a best guess at the various fields based on the names and data.

* X: longitude
* Y: latitude
* OBJECTID: accident ID; integer, 1 through 74305
* ROADSOFTID: 7-digit number associated with roadsoft, possibly [this](http://roadsoft.org/)
* BIKE: was a bike involved? yes or no
* CITY: all are Grand Rapids
* COUNTY: all are Kent County
* CRASHDATE: date of accident, formatted year, month, date
* CRASHSEVER: crash severity, categorical options shown below
* CRASHTYPE: crash type, categorical options shown below
* WORKZNEACT: was crash in an active workzone? categorical options shown below
* WORKZNECLO: nearness to workzone, categorical options shown below
* WORKZNETYP: type of workzone, categorical options shown below  
* CTRLMILEPT: unclear, float between 0 and ~16.69
* CTRLSECT: unclear, float between 0 and 41843 
* DAYOFMONTH: day of month, integer 1-31
* DAYOFWEEK: day of week, categorical  
* ANIMAL: were animals involved, categorical (e.g. wildlife not pets)
* D1COND: appears to be driver 1 condition, categorical shown below; drinking separate category 
* D1DRINKIN: alcohol use for driver 1, yes or no
* D1HAZACT: driver 1 hazardous action, categorical list shown below
* D1INJURY: driver 1 injury, letter corresponding with defined category shown below  
* D1INTENT: driver 1 intended action, categorical list below 
* D2COND: driver 2 condition, categorical shown below; drinking separate category  
* D2DRINKIN: alcohol use for driver 2, yes or no
* D2HAZACT: driver 2 hazardous action, categorical list shown below 
* D2INJURY: driver 2 injury, letter corresponding with defined category shown below   
* D2INTENT: driver 2 intended action, categorical list below 
* D3COND: driver 3 condition, categorical shown below; drinking separate category  
* D3DRINKIN: alcohol use for driver 3, yes or no 
* D3HAZACT: driver 3 hazardous action, categorical list shown below  
* D3INJURY: driver 3 injury, letter corresponding with defined category shown below 
* D3INTENT: driver 3 intended action, categorical list below  
* DRINKING 
* DRIVER1AGE 
* DRIVER1SEX 
* DRIVER2AGE 
* DRIVER2SEX 
* DRIVER3AGE 
* DRIVER3SEX 
* EMRGVEH 
* FARMEQUIP 
* FLEEINGSIT 
* FWSEGID 
* GRTINJSEVE 
* HITANDRUN 
* HOUR 
* INTERNAME 
* LIGHTING 
* MDOTREG 
* MILEPOINT 
* MONTH 
* MOTORCYCLE 
* NOATYPEINJ 
* NOBTYPEINJ 
* NOCTYPEINJ 
* NONTRAFFIC 
* NUMOFINJ 
* NUMOFKILL 
* NUMOFOCCUP 
* NUMOFUNINJ 
* NUMOFVEHIC 
* ORV - off road vehicle?
* PEDESTRIAN 
* PRNAME 
* PRNO 
* REFDIR 
* REFDIST 
* ROUTECLASS 
* ROUTENUM 
* SCHOOLBUS 
* SNOWMOBILE 
* SPDLMTPOST - speed limit posted? 
* SPEEDLIMIT - speed limit
* SURFCOND 
* TRAFCTLDEV 
* TRAIN 
* TRUCKBUS 
* TRUNKLINE 
* UD10NUM
* V1DEFECT 
* V1DAMAGE 
* V1HARMEVT1 
* V1HARMEVT2 
* V1HARMEVT3 
* V1HARMEVT4 
* V1MSTHARME 
* V1SPECCAT 
* V1TRAILER 
* V1VIOLATOR 
* V1WIMPCTPT 
* V2DEFECT 
* V2DAMAGE 
* V2HARMEVT1
* V2HARMEVT2 
* V2HARMEVT3 
* V2HARMEVT4 
* V2MSTHARME 
* V2SPECCAT 
* V2TRAILER 
* V2VIOLATOR 
* V2WIMPCTPT 
* V3DEFECT 

V3 = Vehicle 3?
* V3DAMAGE 
* V3HARMEVT1 
* V3HARMEVT2 
* V3HARMEVT3 
* V3HARMEVT4 
* V3MSTHARME 
* V3SPECCAT 
* V3TRAILER 
* V3VIOLATOR 
* V3WIMPCTPT 
* VEH1DIR 
* VEH1TYPE 
* VEH1USE 
* VEH2DIR 
* VEH2TYPE 
* VEH2USE
* VEH3DIR 
* VEH3TYPE 
* VEH3USE 
* WEATHER 
* WHEREONRD 
* YEAR 
* RDCITYTWP 
* ROAD_USER1 
* ROAD_USER2 
* ROAD_USER3 
* ROAD_USER4 
* RDLEGALSYS 
* RDLGLCODE 
* RDNFC 
* RDNFCCODE 
* RDNUMLANES 
* RDSUBTYPDS
* RDSUBTYPE 
* RDSURFTYPE 
* RDUSRINVID 
* RDWIDTH 
* FRAMEWORK 

## 2. Importing Libraries <a class="anchor" id="header2"></a>

Importing libraries to support data cleaning and exploration.

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

## 3. Importing the Data <a class="anchor" id="header3"></a>

In [5]:
#importing the data
crash_df = pd.read_csv('CGR_Crash_Data.csv')

In [6]:
#checking data shape
crash_df.shape

(74309, 142)

## 4. Exploring the Data Composition <a class="anchor" id="header4"></a>

Data set has 74,309 rows and 142 variables.

In [7]:
#previewing first 5 rows
crash_df.head()

X          Y  OBJECTID  ROADSOFTID BIKE          CITY COUNTY  \
0 -85.650003  42.919854         1     2589528   No  Grand Rapids   Kent   
1 -85.625665  42.924710         2     2593183   No  Grand Rapids   Kent   
2 -85.655282  43.000972         3     2582102   No  Grand Rapids   Kent   
3 -85.643314  42.928172         4     2579820   No  Grand Rapids   Kent   
4 -85.665571  42.968854         5     2594624   No  Grand Rapids   Kent   

    CRASHDATE            CRASHSEVER       CRASHTYPE  ... RDLGLCODE  RDNFC  \
0  2008/06/16  Property Damage Only         Backing  ...         5  Local   
1  2008/08/30  Property Damage Only    Fixed Object  ...         5  Local   
2  2008/02/13  Property Damage Only  Other Driveway  ...         5  Local   
3  2008/01/25  Property Damage Only  Angle Straight  ...         5  Local   
4  2008/09/26  Property Damage Only         Backing  ...         4  Local   

  RDNFCCODE  RDNUMLANES        RDSUBTYPDS  RDSUBTYPE RDSURFTYPE RDUSRINVID  \
0         7           2  Asphalt-Standard         35    Asphalt              
1         7           2  Asphalt-Standard         35    Asphalt              
2         7           2  Asphalt-Standard         35    Asphalt              
3         7           2  Asphalt-Standard         35    Asphalt              
4         7           2  Asphalt-Standard         35    Asphalt              

  RDWIDTH FRAMEWORK  
0    26.0        17  
1    26.0        17  
2    29.0        17  
3    27.0        17  
4    30.0        17  

[5 rows x 142 columns]

In [8]:
#previewing last 5 rows
crash_df.tail()

X          Y  OBJECTID  ROADSOFTID BIKE          CITY COUNTY  \
74304 -85.688410  42.997895     74305     2558829   No  Grand Rapids   Kent   
74305 -85.661670  42.941360     74306     2574652   No  Grand Rapids   Kent   
74306 -85.661243  42.939724     74307     2563322   No  Grand Rapids   Kent   
74307 -85.584837  42.971042     74308     2568387   No  Grand Rapids   Kent   
74308 -85.676419  42.984836     74309     2558314   No  Grand Rapids   Kent   

        CRASHDATE CRASHSEVER        CRASHTYPE  ... RDLGLCODE  \
74304  2017/03/20      Fatal   Other Driveway  ...         4   
74305  2017/11/18      Fatal       Pedestrian  ...         4   
74306  2017/06/02      Fatal  Side-Swipe Same  ...         5   
74307  2017/08/23      Fatal       Pedestrian  ...         3   
74308  2017/03/18      Fatal   Angle Straight  ...         1   

                          RDNFC RDNFCCODE  RDNUMLANES        RDSUBTYPDS  \
74304  Other Principal Arterial         3           2  Asphalt-Standard   
74305            Minor Arterial         4           2  Asphalt-Standard   
74306                     Local         7           2  Asphalt-Standard   
74307                     Local         7           2         Undefined   
74308           Major Collector         5           2         Composite   

       RDSUBTYPE RDSURFTYPE RDUSRINVID RDWIDTH FRAMEWORK  
74304         35    Asphalt               42.0        17  
74305         35    Asphalt               42.0        17  
74306         35    Asphalt               30.0        17  
74307         30  Undefined               26.1        17  
74308         36  Composite               46.0        17  

[5 rows x 142 columns]

In [9]:
#previewing sample of data
crash_df.sample(10)

X          Y  OBJECTID  ROADSOFTID BIKE          CITY COUNTY  \
65221 -85.702881  42.965630     65222     2762964   No  Grand Rapids   Kent   
4906  -85.672149  42.966307      4907     2579473   No  Grand Rapids   Kent   
46564 -85.705201  42.965249     46565     2716904   No  Grand Rapids   Kent   
4200  -85.654432  42.967359      4201     2594418   No  Grand Rapids   Kent   
71583 -85.666738  42.927566     71584     2560804   No  Grand Rapids   Kent   
65293 -85.693086  42.950203     65294     2770912   No  Grand Rapids   Kent   
74138 -85.639613  42.953717     74139     2572305   No  Grand Rapids   Kent   
58111 -85.584369  42.925550     58112     2773779   No  Grand Rapids   Kent   
48015 -85.688257  42.992525     48016     2707280   No  Grand Rapids   Kent   
28367 -85.707861  42.963221     28368     2677622   No  Grand Rapids   Kent   

        CRASHDATE            CRASHSEVER               CRASHTYPE  ...  \
65221  2016/07/11                Injury              Angle Turn  ...   
4906   2008/01/26  Property Damage Only          Angle Straight  ...   
46564  2014/07/27  Property Damage Only              Angle Turn  ...   
4200   2008/09/13  Property Damage Only          Angle Straight  ...   
71583  2017/05/10  Property Damage Only  Misc. Multiple Vehicle  ...   
65293  2016/11/13                Injury              Pedestrian  ...   
74138  2017/10/31                Injury              Pedestrian  ...   
58111  2016/12/15  Property Damage Only       Rear End Straight  ...   
48015  2014/02/11                Injury          Angle Straight  ...   
28367  2012/10/23  Property Damage Only    Misc. Single Vehicle  ...   

      RDLGLCODE                     RDNFC RDNFCCODE  RDNUMLANES  \
65221         1                Interstate         1           1   
4906          4  Other Principal Arterial         3           2   
46564         1                Interstate         1           1   
4200          4           Major Collector         5           2   
71583         4  Other Principal Arterial         3           2   
65293         4            Minor Arterial         4           2   
74138         4                     Local         7           2   
58111         1  Other Principal Arterial         3           2   
48015         4  Other Principal Arterial         3           2   
28367         1                Interstate         1           2   

              RDSUBTYPDS  RDSUBTYPE RDSURFTYPE RDUSRINVID RDWIDTH FRAMEWORK  
65221  Concrete-Standard         37   Concrete                0.0        17  
4906   Concrete-Standard         37   Concrete               56.0        17  
46564  Concrete-Standard         37   Concrete                0.0        17  
4200    Asphalt-Standard         35    Asphalt               35.0        17  
71583   Asphalt-Standard         35    Asphalt               40.0        17  
65293  Concrete-Standard         37   Concrete               50.0        17  
74138   Asphalt-Standard         35    Asphalt               34.0        17  
58111  Concrete-Standard         37   Concrete                0.0        17  
48015   Asphalt-Standard         35    Asphalt               42.0        17  
28367  Concrete-Standard         37   Concrete                0.0        17  

[10 rows x 142 columns]

In [10]:
print(crash_df.columns.tolist())

['X', 'Y', 'OBJECTID', 'ROADSOFTID', 'BIKE', 'CITY', 'COUNTY', 'CRASHDATE', 'CRASHSEVER', 'CRASHTYPE', 'WORKZNEACT', 'WORKZNECLO', 'WORKZNETYP', 'CTRLMILEPT', 'CTRLSECT', 'DAYOFMONTH', 'DAYOFWEEK', 'ANIMAL', 'D1COND', 'D1DRINKIN', 'D1HAZACT', 'D1INJURY', 'D1INTENT', 'D2COND', 'D2DRINKIN', 'D2HAZACT', 'D2INJURY', 'D2INTENT', 'D3COND', 'D3DRINKIN', 'D3HAZACT', 'D3INJURY', 'D3INTENT', 'DRINKING', 'DRIVER1AGE', 'DRIVER1SEX', 'DRIVER2AGE', 'DRIVER2SEX', 'DRIVER3AGE', 'DRIVER3SEX', 'EMRGVEH', 'FARMEQUIP', 'FLEEINGSIT', 'FWSEGID', 'GRTINJSEVE', 'HITANDRUN', 'HOUR', 'INTERNAME', 'LIGHTING', 'MDOTREG', 'MILEPOINT', 'MONTH', 'MOTORCYCLE', 'NOATYPEINJ', 'NOBTYPEINJ', 'NOCTYPEINJ', 'NONTRAFFIC', 'NUMOFINJ', 'NUMOFKILL', 'NUMOFOCCUP', 'NUMOFUNINJ', 'NUMOFVEHIC', 'ORV', 'PEDESTRIAN', 'PRNAME', 'PRNO', 'PUBLICPROP', 'REFDIR', 'REFDIST', 'ROUTECLASS', 'ROUTENUM', 'SCHOOLBUS', 'SNOWMOBILE', 'SPDLMTPOST', 'SPEEDLIMIT', 'SURFCOND', 'TRAFCTLDEV', 'TRAIN', 'TRUCKBUS', 'TRUNKLINE', 'UD10NUM', 'V1DEFECT', 'V

In [25]:
crash_df.describe()

X             Y      OBJECTID    ROADSOFTID    CTRLMILEPT  \
count  74309.000000  74309.000000  74309.000000  7.430900e+04  74309.000000   
mean     -85.653561     42.956494  37155.000000  2.670271e+06      2.404829   
std        0.034017      0.029451  21451.304914  6.471696e+04      4.337907   
min      -85.751682     42.883679      1.000000  2.558094e+06      0.000000   
25%      -85.677046     42.932320  18578.000000  2.613603e+06      0.000000   
50%      -85.660144     42.962892  37155.000000  2.671346e+06      0.000000   
75%      -85.635768     42.974368  55732.000000  2.726270e+06      2.895000   
max      -85.568651     43.028975  74309.000000  2.781429e+06     16.691000   

           CTRLSECT    DAYOFMONTH    DRIVER1AGE    DRIVER2AGE    DRIVER3AGE  \
count  74309.000000  74309.000000  74309.000000  74309.000000  74309.000000   
mean   16780.303288     15.659436    154.512253    340.889152    936.005141   
std    20189.023052      8.764828    315.686635    446.285033    237.750798   
min        0.000000      1.000000      0.000000      0.000000      0.000000   
25%        0.000000      8.000000     24.000000     29.000000    999.000000   
50%        0.000000     16.000000     36.000000     49.000000    999.000000   
75%    41051.000000     23.000000     56.000000    999.000000    999.000000   
max    41843.000000     31.000000    999.000000    999.000000    999.000000   

       ...      ROUTENUM    SPEEDLIMIT       UD10NUM          YEAR  \
count  ...  74309.000000  74309.000000  7.430900e+04  74309.000000   
mean   ...    173.594921     38.634566  7.785196e+06   2012.777106   
std    ...    323.917909     17.106006  2.307781e+06      2.905732   
min    ...      0.000000      0.000000  1.000000e+06   2008.000000   
25%    ...      0.000000     25.000000  7.413624e+06   2010.000000   
50%    ...     11.000000     30.000000  8.372076e+06   2013.000000   
75%    ...    131.000000     45.000000  9.139631e+06   2015.000000   
max    ...    999.000000     99.000000  9.999999e+06   2017.000000   

          RDLGLCODE     RDNFCCODE    RDNUMLANES     RDSUBTYPE       RDWIDTH  \
count  74309.000000  74309.000000  74309.000000  74309.000000  74309.000000   
mean       3.022770      3.656058      2.351034     35.601879     25.549317   
std        1.592737      1.835688      0.939998      1.850789     20.026456   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        1.000000      3.000000      2.000000     35.000000      0.000000   
50%        4.000000      3.000000      2.000000     35.000000     30.000000   
75%        4.000000      4.000000      2.000000     37.000000     42.000000   
max        7.000000      7.000000      6.000000     38.000000     80.000000   

       FRAMEWORK  
count    74309.0  
mean        17.0  
std          0.0  
min         17.0  
25%         17.0  
50%         17.0  
75%         17.0  
max         17.0  

[8 rows x 33 columns]

Exploring options and corresponding count for categorical columns:

In [16]:
#bike involvement
crash_df['BIKE'].value_counts()

No     73383
Yes      926
Name: BIKE, dtype: int64

In [17]:
#city crash occurred in, all Grand Rapids
crash_df['CITY'].value_counts()

Grand Rapids    74309
Name: CITY, dtype: int64

In [19]:
#count of crash, all Kent
crash_df['COUNTY'].value_counts()

Kent    74309
Name: COUNTY, dtype: int64

In [14]:
#crash severity
crash_df['CRASHSEVER'].value_counts()

Property Damage Only    60761
Injury                  13443
Fatal                     105
Name: CRASHSEVER, dtype: int64

In [15]:
#type of crash
crash_df['CRASHTYPE'].value_counts()

Rear End Straight                                 21746
Side-Swipe Same                                   11214
Angle Straight                                     8703
Fixed Object                                       7276
Misc. Multiple Vehicle                             4141
Angle Turn                                         2907
Backing                                            2179
Parking                                            2166
Angle Driveway                                     2135
Side-Swipe Opposite                                1583
Head-On Left-Turn Not Associated with Driveway     1576
Rear End Driveway                                  1211
Pedestrian                                         1044
Head-on                                            1011
Animal                                              910
Bicycle                                             908
Other Driveway                                      776
Rear End Left Turn                              

In [11]:
#active workzone and type of work occuring
crash_df['WORKZNEACT'].value_counts()

Uncoded & Errors              72707
Other                           750
Work on Shoulder / Median       637
Lane Closure                    177
Lane Shift / Crossover           32
Intermittent / Moving Work        6
Name: WORKZNEACT, dtype: int64

In [13]:
#closeness to the workzone
crash_df['WORKZNECLO'].value_counts()

Uncoded & Errors                                      74048
Between the First and Last Work Zone Warning Signs      193
No Warning Signs                                         43
Before the First Work Zone Warning Sign                  25
Name: WORKZNECLO, dtype: int64

In [20]:
#workzone type (contruction/maintenance or utility)
crash_df['WORKZNETYP'].value_counts()

Uncoded & Errors               73197
Construction or Maintenance     1046
Utility                           66
Name: WORKZNETYP, dtype: int64

In [29]:
#day of month 
crash_df['DAYOFMONTH'].value_counts()

14    2666
1     2619
10    2619
22    2595
21    2558
19    2516
23    2494
11    2490
13    2487
15    2482
18    2474
3     2460
9     2441
4     2431
12    2425
8     2415
28    2412
2     2406
7     2405
17    2401
20    2398
6     2381
24    2370
26    2368
16    2367
27    2354
5     2250
29    2202
30    2195
25    2166
31    1462
Name: DAYOFMONTH, dtype: int64

In [26]:
#day of week
crash_df['DAYOFWEEK'].value_counts()

Friday       12546
Wednesday    11929
Thursday     11439
Tuesday      10974
Monday       10442
Saturday      9739
Sunday        7240
Name: DAYOFWEEK, dtype: int64

In [30]:
#animal involvement
crash_df['ANIMAL'].value_counts()

No, Uncoded & Errors    73446
Deer                      854
Other                       7
Turkey                      2
Name: ANIMAL, dtype: int64

In [31]:
#driver 1 condition
crash_df['D1COND'].value_counts()

Appeared Normal        57431
Uncoded & Errors       10106
Unknown                 5552
Other                    429
Fatigue / Asleep         323
Sick                     230
Medication               110
Emotional                104
Physically Disabled       24
Name: D1COND, dtype: int64

In [32]:
#driver 1 condition
crash_df['D1DRINKIN'].value_counts()

No     71627
Yes     2682
Name: D1DRINKIN, dtype: int64

In [33]:
#driver 1 hazardous actions
crash_df['D1HAZACT'].value_counts()

None                    22900
Fail to Stop ACD        14251
Failed to Yield          8597
Speed too Fast           6283
Other                    5286
Unknown                  4147
Disobeyed TCD            2572
Improper Lane Use        2420
Improper Backing         2134
Careless Driving         2095
Improper Turn            1378
Reckless Driving          681
Improper Pass             569
Drove Left of Center      399
Uncoded & Errors          210
Improper Signal           166
Drove Wrong Way           131
Speed too Slow             90
Name: D1HAZACT, dtype: int64

In [39]:
#driver 2 injury
crash_df['D2INJURY'].value_counts()

O-No Injury                    45388
Uncoded & Errors               22866
C-Possible injury               4811
B-Non-incapacitating Injury      932
A-Incapacitating Injury          283
K-Fatal injury                    29
Name: D2INJURY, dtype: int64

In [40]:
#driver 2 intented action 
crash_df['D2INTENT'].value_counts()

Going Straight                             26805
Uncoded & Errors                            9992
Stopped on Road                             8949
Slowing or Stopped on Road                  8112
Parked                                      6142
Turning Left                                3618
Unknown                                     1776
Turning Right                               1703
Changing Lanes                              1649
Backing                                     1321
Starting up on Road                         1042
Entering Road                                527
Crossing at Intersection                     511
Overtaking or Passing                        449
Avoiding Vehicle from the front or back      361
Crossing Midblock                            216
Leaving Parking                              181
Other                                        123
Avoiding Vehicle at an angle                 122
Slowing or Stopped, Other                    115
In Prior Crash      

In [41]:
#driver 2 condition
crash_df['D2COND'].value_counts()

Appeared Normal        47110
Uncoded & Errors       19549
Unknown                 7361
Other                     96
Sick                      72
Fatigue / Asleep          47
Emotional                 46
Medication                23
Physically Disabled        5
Name: D2COND, dtype: int64

In [42]:
#driver 2 alcohol involvement
crash_df['D2DRINKIN'].value_counts()

No                  63979
Uncoded & Errors     9808
Yes                   522
Name: D2DRINKIN, dtype: int64

In [43]:
#driver 2 hazardous actions
crash_df['D2HAZACT'].value_counts()

None                    42033
Uncoded & Errors        10177
Fail to Stop ACD         5786
Unknown                  4377
Failed to Yield          3437
Other                    2033
Improper Lane Use        1347
Disobeyed TCD            1095
Improper Backing         1066
Speed too Fast            821
Careless Driving          593
Improper Turn             561
Improper Pass             373
Drove Left of Center      223
Reckless Driving          172
Improper Signal            98
Drove Wrong Way            59
Speed too Slow             58
Name: D2HAZACT, dtype: int64

In [44]:
#driver 2 injury
crash_df['D2INJURY'].value_counts()

O-No Injury                    45388
Uncoded & Errors               22866
C-Possible injury               4811
B-Non-incapacitating Injury      932
A-Incapacitating Injury          283
K-Fatal injury                    29
Name: D2INJURY, dtype: int64

In [45]:
#drier 2 intended action
crash_df['D2INTENT'].value_counts()

Going Straight                             26805
Uncoded & Errors                            9992
Stopped on Road                             8949
Slowing or Stopped on Road                  8112
Parked                                      6142
Turning Left                                3618
Unknown                                     1776
Turning Right                               1703
Changing Lanes                              1649
Backing                                     1321
Starting up on Road                         1042
Entering Road                                527
Crossing at Intersection                     511
Overtaking or Passing                        449
Avoiding Vehicle from the front or back      361
Crossing Midblock                            216
Leaving Parking                              181
Other                                        123
Avoiding Vehicle at an angle                 122
Slowing or Stopped, Other                    115
In Prior Crash      

## References

* [Showing all columns as a list](https://www.statology.org/pandas-show-all-columns/)